testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -601,6 +601,145 @@ public void testExplainAnalyze()
+
+        assertQuery(
+                selectMaterializedViewsSql(),
+                getMaterializedViewsResultRows(materializedViewName, true, otherMaterializedViewName));
+
+        // test filtering by columns
+        assertQuery(
+                selectMaterializedViewsSql(format("catalog_name = '%s'", getSession().getCatalog().orElseThrow())),
+                getMaterializedViewsResultRows(materializedViewName, true, otherMaterializedViewName));
+
+        assertQuery(
+                selectMaterializedViewsSql(
+                        format("catalog_name = '%s'", getSession().getCatalog().orElseThrow()),
+                        format("schema_name = '%s'", otherSchemaName)),
+                getMaterializedViewsResultRow(otherMaterializedViewName, otherSchemaName, false));
+
+        assertQuery(
+                selectMaterializedViewsSql(
+                        format("catalog_name = '%s'", getSession().getCatalog().orElseThrow()),
+                        format("schema_name = '%s'", getSession().getSchema().orElseThrow()),
+                        format("name = '%s'", materializedViewName)),
+                getMaterializedViewsResultRow(materializedViewName, getSession().getSchema().orElseThrow(), true));
+
+        assertQuery(
+                selectMaterializedViewsSql(format("schema_name LIKE '%%%s%%'", getSession().getSchema().orElseThrow())),
+                getMaterializedViewsResultRow(materializedViewName, getSession().getSchema().orElseThrow(), true));
+
+        assertQuery(
+                selectMaterializedViewsSql(format("name LIKE '%%%s%%'", materializedViewName)),
+                getMaterializedViewsResultRow(materializedViewName, getSession().getSchema().orElseThrow(), true));
+
+        computeActual(format("DROP MATERIALIZED VIEW %s.%s.%s",
+                getSession().getCatalog().orElseThrow(),
+                getSession().getSchema().orElseThrow(),
+                materializedViewName));
+        computeActual(format("DROP MATERIALIZED VIEW %s.%s.%s",
+                getSession().getCatalog().orElseThrow(),
+                otherSchemaName,
+                otherMaterializedViewName));
+
+        assertQueryReturnsEmptyResult(selectMaterializedViewsSql());
+    }
+
+    private String createMaterializedViewSql(String materializedViewName, String schemaName)
+    {
+        return format(
+                "CREATE MATERIALIZED VIEW %s.%s.%s COMMENT 'sarcastic comment' AS %s",
+                getSession().getCatalog().orElseThrow(),
+                schemaName,
+                materializedViewName,
+                materializedViewDefinitionSql());
+    }
+
+    private String getMaterializedViewsResultRow(
+            String materializedViewName,
+            String schemaName,
+            boolean isMaterializedViewFresh)
+    {
+        return format(
+                "VALUES ('%s', '%s', '%s', '%b', 'user', 'sarcastic comment', '%s')",
```

Comment:
We should get user from `String user = getSession().getUser()` instead of hard-coding 'user' in the helper methods